CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -256,8 +256,13 @@ public extension WorkspaceClient {
             // if a file/folder with the same name exists, add "copy" to the end
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
-                let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let previousName = (fileUrl.lastPathComponent as NSString).deletingPathExtension
```

Comment:
if you change the order then you not need to cast to NSString:

```suggestion
                let previousName = fileUrl.deletingPathExtension().lastPathComponent
```

Reply:
I tried the suggested way but when trying to duplicate. It just hangs and doesn't do anything.